flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/PartitionableSourceTest.scala

```diff
@@ -148,6 +148,16 @@ class PartitionableSourceTest(val sourceFetchPartitions: Boolean, val useCatalog
       "select name from PartitionableAndFilterableTable " +
         "where part1 = 'A' and part2 > 1 and id > 1")
   }
+
+  @Test
+  def testRandCondition(): Unit = {
+    util.verifyRelPlan("SELECT * FROM PartitionableTable WHERE rand(1) < 0.001")
+  }
+
+  @Test
+  def testRandCondition2(): Unit = {
+    util.verifyRelPlan("SELECT * FROM PartitionableTable WHERE rand(part2) < 0.001")
+  }
```

Comment:
These are the same tests as tests you added in `PushPartitionIntoLegacyTableSourceScanRuleTest`. I think there is no need to add these tests. WDYT?

Reply:
I mainly follow the suggestion as here: https://github.com/apache/flink/pull/12966#pullrequestreview-455454350
Added the test in the rule test `PushPartitionIntoLegacyTableSourceScanRuleTest` and  `PartitionableSourceTest`